src/Publisher.php

```diff
@@ -14,34 +14,52 @@
 namespace Symfony\Component\Mercure;
 
 use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mercure\Token\CallableTokenProvider;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 /**
  * Publishes an update to the hub.
  *
- * Can be used as a Symfony Messenger handler too.
- *
  * @author Kévin Dunglas <dunglas@gmail.com>
  *
  * @experimental
  */
 final class Publisher implements PublisherInterface
 {
-    private $hubUrl;
-    private $jwtProvider;
+    private $hub;
     private $httpClient;
 
     /**
-     * @param callable(Update $update): string $jwtProvider
+     * @param Hub                      $hub
+     * @param HttpClientInterface|null $httpClient
      */
-    public function __construct(string $hubUrl, callable $jwtProvider, HttpClientInterface $httpClient = null)
+    public function __construct($hub, $httpClient = null)
     {
-        $this->hubUrl = $hubUrl;
-        $this->jwtProvider = $jwtProvider;
+        if (\is_string($hub)) {
```

Comment:
```suggestion
        if (!$hub instanceof Hub) {
```

So we also handle edge cases such as `Stringable` objects etc.